File: leetcode/rust-leetcode/src/3042.count-prefix-and-suffix-pairs-i.rs
impl Solution {
    pub fn count_prefix_suffix_pairs(words: Vec<String>) -> i32 {
        let is_prefix_and_suffix = |i: usize, j: usize| -> bool {
            words[j].len() >= words[i].len() && words[j].starts_with(&words[i]) && words[j].ends_with(&words[i])
        };
        let mut count = 0;
        for i in 0..words.len()-1 {
            for j in i+1..words.len() {
                if is_prefix_and_suffix(i, j) {
                    count += 1;
                }
            }
        }
        count
    }
}

File: leetcode/python/3042.count-prefix-and-suffix-pairs-i.py
class Solution:
    def countPrefixSuffixPairs(self, words: List[str]) -> int:
        def isPrefixSuffix(wordi: int, wordj: int) -> int:
            return len(wordj) >= len(wordi) and wordj.startswith(wordi) and wordj.endswith(wordi)
        ans = 0
        for i in range(len(words)-1):
            for j in range(i+1, len(words)):
                ans += isPrefixSuffix(words[i], words[j])
        return ans

